job-application-tracker-api/src/applications/applications.service.ts

```diff
@@ -0,0 +1,65 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { ApplicationRepository } from './application.repository';
+import { UsersService } from 'src/users/users.service';
+import { ApplicationMapper } from './application-mapper';
+import { CreateApplicationDto } from './dto/create-application.dto';
+import { UpdateApplicationDto } from './dto/update-application.dto';
+
+@Injectable()
+export class ApplicationsService {
+    constructor(private readonly applicationRepository: ApplicationRepository,
+        private readonly userService: UsersService
+    ) { }
+
+    async findAllUserApplications(userId: string) {
+        try {
+            this.userService.findOne(userId);
+        }
+        catch {
+            throw new NotFoundException('no such user');
+        }
+        return this.applicationRepository.findAllByUserId(userId);
+    }
+
+    async findOne(applicationId: string) {
+        const application = await this.applicationRepository.findById(applicationId);
+        if (!application) throw new NotFoundException(' no such application');
+        return ApplicationMapper.toApplicationDto(application);
+    }
+
+    async create(createApplicationDto: CreateApplicationDto) {
+        try {
+            this.userService.findOne(createApplicationDto.userId);
+        }
+        catch {
+            throw new NotFoundException('no such user');
+        }
+
+        const newApplication : CreateApplicationDto = {
+            companyName: createApplicationDto.companyName,
+            position: createApplicationDto.position,
+            states: createApplicationDto.states,
+            userId: createApplicationDto.userId
+        };
```

Comment:
Why would you not send the CreateApplicationDto directly to applicationRepository.create

Reply:
וואלה לא יודע